python/ray/train/torch/torch_checkpoint.py

```diff
@@ -51,7 +51,7 @@ def _encode_data_dict(self, data_dict: dict) -> dict:
         # are in the checkpoint dict can be properly deserialized on the
         # driver side, even if the driver does not have access to a GPU device.
         _buffer = io.BytesIO()
-        torch.save(data_dict, _buffer, pickle_module=ray.cloudpickle)
+        torch.save(data_dict, _buffer, pickle_module=ray.cloudpickle, pickle_protocol=5)
```

Comment:
Can we add a simple test for this?

Reply:
Yeah sure. I'll add one

Reply:
For Python version compatibility I think we should use `DEFAULT_PROTOCOL` instead

https://github.com/ray-project/ray/blob/d0153a394a2f9078a1861e48e764774657464f95/python/ray/cloudpickle/cloudpickle.py#L86

Reply:
I see. That make sense!